homeassistant/config_entries.py

```diff
@@ -817,11 +817,22 @@ def async_get_options_flow(config_entry: ConfigEntry) -> "OptionsFlow":
         raise data_entry_flow.UnknownHandler
 
     @callback
-    def _abort_if_unique_id_configured(self) -> None:
+    def _abort_if_unique_id_configured(
+        self, update_entry: Dict[Any, Any] = None
+    ) -> None:
         """Abort if the unique ID is already configured."""
         if self.unique_id is None:
             return
 
+        if update_entry is not None:
```

Comment:
Let's merge this for loop with the self._async_current_ids() check. Move `if update_entry is not None` inside the unique ID check.

Reply:
We could probably also remove `async_current_ids`, it's not being used ?

Reply:
It is used by Hue?

https://github.com/home-assistant/home-assistant/blob/b63dd611b76fb4a1c7fec1a8dbccb2965c917dc0/homeassistant/components/hue/config_flow.py#L78-L80

Reply:
Addressed the comments, except for the removal of `async_current_ids`, that one is still in. I think Hue is displaying a fair use-case of that method.